lightning/src/chain/onchaintx.rs

```diff
@@ -394,10 +428,26 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 			preprocessed_requests.push(req);
 		}
 
+		loop {
+			let pop_height = if let Some(first_entry) = self.locktimed_packages.iter().next() {
+				if *first_entry.0 <= height + 1 {
+					*first_entry.0
+				} else { break; }
+			} else { break; };
+			log_trace!(logger, "Restoring delayed claim of package(s) at their timelock at {}.", pop_height);
+			preprocessed_requests.append(&mut self.locktimed_packages.remove(&pop_height).unwrap());
+		}
```

Comment:
No, I think you're very right - I was thking that because we don't have a `drain` we shouldn't as we'll end up copying a tone, but it doesn't actually matter because we can `drain` (or `append`-from) the inner `Vec`. It does make it much more readable.